add Android NDK paths and plugin import options for arm64 architecture

nutcas3 opened the pull request:
Triies to fix #1 missing libs

nutcas3 committed:
add Android NDK paths and plugin import options for arm64 architecture

nutcas3 committed:
Fix Android Qt build: add explicit build and staging for qtforandroid and qtfreetype libs

- Add build commands for src/plugins/platforms/android and src/platformsupport/fontdatabases/freetype
- Stage libqtforandroid.a as libplugins_platforms_qtforandroid.a and libqtfreetype.a for linker
- Add missing include paths and NDK platform flags to Android Qt configuration
- Skip qtwebengine to streamline build and avoid unnecessary dependencies

Fixes linker errors: unable to find library -lplugins_platforms_qtforandroid and -lqtfreetype

naftalimurgor commented:
Hello @nutcas3 :wave: 

Thank you so much for this fix, let me check out the changes, test, then will merge, a few pointers:
1. `make -j $(nproc)` should exit successfully- we need this step for `libbitgesell.so` lib which will be the JNI lib that the `apk` will bundle/build will be part of
2. We are on tag version `0.1.13` - which has `Qt 51.5`

Everything looks good, just need to test then we can proceed to the last step- Got  a patch for the make process to bundle the Android apk if missing in `0.1.13` tag

nutcas3 commented:
Sure seen the pointers 
Just tell me how it works then I will see the other parts

nutcas3 committed:
Fix Qt Android linker errors by adding missing plugin and Freetype library paths

- Added platform plugin library paths for Android to resolve -lplugins_platforms_qtforandroid error
- Added Freetype library and include paths to resolve -lqtfreetype linker error
- Ensured build configuration can properly locate required Qt libraries for Android builds

nutcas3 commented:
Hello @naftalimurgor check on the latest one and send me the error if any

naftalimurgor commented:
Hello @nutcas3 I appreciate the contribution,
-  Have the   the changes above been tested? 
- Tried setting up the docker image and following through the steps? 
- Also check out the 0.1.9 release tag as that's the one the focus is currently on.

nutcas3 commented:
I will do the test today for you I will send the benchmarks or the results

naftalimurgor commented:
sounds great. Looking forward to!
